test/example-types/facet.js

```diff
@@ -434,5 +435,82 @@ describe('facet', () => {
         ],
       })
     })
+
+    describe('should always include checked values', () => {
+      let Data = [
+        { _id: 1, name: '1' },
+        { _id: 2, name: '2' },
+        { _id: 3, name: '3' },
+        { _id: 4, name: '4' },
+      ]
+      let mongoIdData = [
+        { _id: '5e9dbd76e991760021124966', name: 'Automation' },
+        { _id: '5cde2658dc766b0030c67dae', name: 'Fowlkes (MO)' },
+        { _id: '5d1ca49436e1d20038f8c84f', name: 'Customer Experience' },
+        { _id: '5ce30b403aa154002d01b9ed', name: 'Government Division' },
+      ]
+      let node = {
+        key: 'id',
+        field: '_id',
+        type: 'facet',
+        label: {
+          collection: null,
+          foreignField: '_id',
+          fields: ['name'],
+        },
+        mode: 'include',
+        optionsFilter: '',
+        size: 2,
+      }
+      it('when there are missed values', async () => {
+        ;(node.label.collection = Data), (node.values = [4])
+
+        let result = await facet.result(node, agg => mingo.aggregate(Data, agg))
+        let ids = _.map(({ name }) => _.toString(name), result.options)
+        expect(result.options.length).to.equal(2)
+        expect(_.includes('4', ids)).to.be.true
+      })
+      it('when there is no missed value', async () => {
+        ;(node.label.collection = Data), (node.values = [1])
+        let result = await facet.result(node, agg => mingo.aggregate(Data, agg))
+        let ids = _.map(({ name }) => _.toString(name), result.options)
+        expect(result.options.length).to.equal(2)
+        expect(_.includes('1', ids)).to.be.true
+      })
+      it('when there are missed values and isMongoId: true', async () => {
```

Comment:
`'should correctly include missed values when isMongoId is true'`

Reply:
since we have described ('should always include checked values in result'. and this test case is one of a different condition, normally we only use "when ....." I could change it to "when missing values  is expected and  isMongoId is true"

Reply:
![image](https://user-images.githubusercontent.com/10645051/92290274-94386e80-eee1-11ea-8697-cef0192ceaa8.png)